src/spikeinterface/preprocessing/filter.py

```diff
@@ -65,24 +65,21 @@ def __init__(
         ftype="butter",
         filter_mode="sos",
         margin_ms=5.0,
+        add_reflect_padding=False,
         coeff=None,
         dtype=None,
     ):
         import scipy.signal
 
         assert filter_mode in ("sos", "ba")
-        sf = recording.get_sampling_frequency()
+        fs = recording.get_sampling_frequency()
         if coeff is None:
             assert btype in ("bandpass", "highpass")
             # coefficient
-            if btype in ("bandpass", "bandstop"):
-                assert len(band) == 2
-                Wn = [e / sf * 2 for e in band]
-            else:
-                Wn = float(band) / sf * 2
-            N = filter_order
             # self.coeff is 'sos' or 'ab' style
-            filter_coeff = scipy.signal.iirfilter(N, Wn, analog=False, btype=btype, ftype=ftype, output=filter_mode)
+            filter_coeff = scipy.signal.iirfilter(
+                filter_order, band, fs=fs, analog=False, btype=btype, ftype=ftype, output=filter_mode
+            )
```

Comment:
Fair enough. I was thinking that `fs` was introduced lately but it was 5 yeras ago!